### DISCUSSION TITLE
Make Algolia optional

### DISCUSSION THREAD

**Discussion by donmhico** [WebDevStudios/nextjs-wordpress-starter#229] [Ideas]
I noticed that if the Algolia env vars are not set ( `NEXT_PUBLIC_ALGOLIA_INDEX_NAME`, `NEXT_PUBLIC_ALGOLIA_APPLICATION_ID`, `NEXT_PUBLIC_ALGOLIA_SEARCH_ONLY_KEY` ) the site doesn't load. I get this error

```
Server Error

Error: Error serializing `.algolia.indexName` returned from `getStaticProps` in "/".
Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value.
```

In my own opinion, as a _starter_, I think the lesser we need to setup the better. 

We can change this block of code

https://github.com/WebDevStudios/nextjs-wordpress-starter/blob/1ed3c99cd5115ef8239ecd89f323173c289bf357/api/algolia/connector.js#L4-L6

to

```js
// Define env vars.
export const algoliaIndexName = process.env.NEXT_PUBLIC_ALGOLIA_INDEX_NAME ?? null
export const algoliaSearchKey = process.env.NEXT_PUBLIC_ALGOLIA_SEARCH_ONLY_KEY ?? null
export const algoliaAppId = process.env.NEXT_PUBLIC_ALGOLIA_APPLICATION_ID ?? null
```

This change will make the Algolia related env vars optional, at least until the user actually wants to use the Algolia Search component.

**Comment by gregrickaby** [comment]
@donmhico I'm fine with this change. Submit a PR and I'll get it merged in.
